File: Java/Assignment3/Assignment3_2.java

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// file name   : Assignment3_2.java
// Description : write a program which accept one number from user and print  even factors of that numbers on screen.
// Author      : Aniket Devaba Markad                        
// Date        : 28/11/2025 
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class number
{

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name : DisplayEvenFactor 
// Description   : This function is used to print even factor of number numbers. 
// Author        : Aniket Devaba Markad 
// Date          : 28/11/2025 
// 
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

    void DisplayEvenFactor(int iNo)
    {
        int iCnt = 0;

        if(iNo < 0)
        {
            iNo = -(iNo);     //updator
        }
        System.out.println("Even factor of "+iNo+" : " );

        for(iCnt = 1;iCnt <= iNo/2; iCnt++)
        {
            if(iNo % iCnt == 0)
            {
                if(iCnt % 2 == 0)
                {
                    System.out.println(iCnt);
                }
            }
        }
    }
}

public class Assignment3_2
{
    public static void main(String a[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.println("Enter the number");
        iValue = sobj.nextInt();

        number nobj = new number();
        nobj.DisplayEvenFactor(iValue);
    }
}

File: CPP/Assignment3_2.cpp
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// file name    : Assignment3_2.cpp
// Description  : Write a program which accept number from user and display all factors of that number.
// Author       : Aniket Devaba Markad
// Date         : 09/12/2025
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name : DisplayFactors
// Description   : This function is used to display factors of number.
// Input         : int
// Output        : void
// Author        : Aniket Devaba Markad
// Date          : 09/12/2025
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

void DisplayFactors(int iNo)
{
    int iCnt = 0;

    if(iNo < 0)
    {
        iNo = -iNo;
    }

    for(iCnt = 1; iCnt <= (iNo / 2); iCnt++)
    {
        if((iNo % iCnt) == 0)
        {
            cout << iCnt << "\n";
        }
    }
}

int main()
{
    int iValue = 0;

    cout << "Enter number:";
    cin >> iValue;

    DisplayFactors(iValue);

    return 0;
}
